src/features/SendAssets/SendAssets.reducer.tsx

```diff
@@ -0,0 +1,153 @@
+import { hexlify, bigNumberify, Arrayish } from 'ethers/utils';
+import { TransactionResponse } from 'ethers/providers';
+import { ValuesType } from 'utility-types';
+
+import {
+  IFormikFields,
+  ITxObject,
+  Asset,
+  TAddress,
+  ITxConfig,
+  ITxReceipt,
+  ISignedTx,
+  ITxHash,
+  ITxType,
+  ExtendedAsset,
+  Network,
+  StoreAccount,
+  TAction
+} from '@types';
+import {
+  getBaseAssetByNetwork,
+  hexWeiToString,
+  bigNumGasPriceToViewableGwei,
+  inputGasPriceToHex
+} from '@services';
+import { makePendingTxReceipt, makeTxConfigFromSignedTx } from '@utils';
+
+import { processFormDataToTx } from './helpers';
+
+interface State {
+  txConfig?: ITxConfig;
+  txReceipt?: ITxReceipt;
+  signedTx?: ISignedTx;
+  send?: boolean;
+}
+
+export type ReducerAction = TAction<ValuesType<typeof sendAssetsReducer.actionTypes>, any>;
+
+// @ts-ignore
+export const initialState: State = { txConfig: {} };
+
+export const sendAssetsReducer = (state: State, action: ReducerAction) => {
+  switch (action.type) {
+    case sendAssetsReducer.actionTypes.FORM_SUBMIT: {
+      const txConfig = handleFormSubmit(action.payload);
```

Comment:
Applies for the whole reducer:
- since the reducer is a pure function without side effects, it no longer 'handles' a flow. The 'handling' is done in the effects and the reducer just transforms and sets data. At first glance most of the transformations can be done directly in the reducer and we don't need a helper function for them. We're testing the reducer. If you really prefer a helper function give it an appropriate name.